test/unit/modules/blocks/utils.js

```diff
@@ -13,268 +13,558 @@
+			expect(() => {
+				testTracker.applyNext();
+			}).to.throw('Cannot apply transaction over the limit: 1');
 		});
 
-		describe('when db query succeeds', () => {
-			describe('and does not return results', () => {
-				it('should set height to 0');
-			});
-
-			describe('and returns results', () => {
-				it('should set height to rows[0].height');
-			});
-
-			it('should set params.limit to realLimit');
-
-			it('should set params.height to height');
-
-			it('should call library.db.query');
-
-			it('should call library.db.query with sql.loadBlocksData with filter');
-
-			it('should call library.db.query with params)');
-
-			describe('when db query succeeds', () => {
-				it('should call callback with error = null');
-
-				it('should call callback with result containing the rows');
-			});
+		it('should return valid log information when reset tracker and call applyNext()', () => {
+			testTracker.reset();
+			testTracker.applyNext();
+			expect(loggerStub.info.args[0][0]).to.equal('Test tracker');
+			expect(loggerStub.info.args[0][1]).to.equal(
+				'100.0 %: applied 1 of 1 transactions'
+			);
 		});
 	});
 
-	describe('getBlockProgressLogger', () => {
-		describe('BlockProgressLogger', () => {
-			it('should set this.target to transactionsCount');
-
-			it(
-				'should set this.step to Math.floor(transactionsCount / logsFrequency);'
+	describe('aggregateBlocksReward', () => {
+		it('should return error when account.get fails', done => {
+			blocksUtilsModule.aggregateBlocksReward(
+				{ address: 'ERRL' },
+				(err, data) => {
+					expect(err).to.equal('Address error stub');
+					expect(data).to.be.undefined;
+					done();
+				}
 			);
+		});
 
-			it('should set this.applied to 0');
-
-			describe('reset function', () => {
-				it('should set this.applied to 0');
-			});
-
-			describe('applyNext function', () => {
-				describe('when this.applied >= this.target', () => {
-					it('should throw error');
-				});
-
-				it('should increment this.applied');
-
-				describe('when this.applied = 1', () => {
-					it('should call this.log');
-				});
-
-				describe('when this.applied = this.target', () => {
-					it('should call this.log');
-				});
-
-				describe('when this.applied %this.step = 1', () => {
-					it('should call this.log');
-				});
-			});
+		it('should return error when account not found', done => {
+			blocksUtilsModule.aggregateBlocksReward(
+				{ address: '0L' },
+				(err, data) => {
+					expect(err).to.equal('Account not found');
+					expect(data).to.be.undefined;
+					done();
+				}
+			);
+		});
 
-			describe('log function', () => {
-				it('should call library.logger.info with msg');
+		it('should return error when library.db.blocks.aggregateBlocksReward fails', done => {
+			blocksUtilsModule.aggregateBlocksReward(
+				{ address: '1L' },
+				(err, data) => {
+					expect(loggerStub.error.args[0][0]).to.contains(
+						"TypeError: Cannot read property '0' of undefined"
```

Comment:
I don't think that is correct behavior.

Reply:
Returning/catching the error when `library.db.blocks.aggregateBlocksReward` fails is the behavior, not the string logged itself.